client/state/sites/updates/reducer.js

```diff
@@ -2,60 +2,61 @@
  * External dependencies
  */
 import { combineReducers } from 'redux';
+import { reduce } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import { createReducer } from 'state/utils';
 
 import {
+	SITE_RECEIVE,
+	SITES_RECEIVE,
+	SITES_UPDATE,
 	SITE_UPDATES_RECEIVE,
 	SITE_UPDATES_REQUEST,
 	SITE_UPDATES_REQUEST_SUCCESS,
 	SITE_UPDATES_REQUEST_FAILURE,
-	SERIALIZE,
-	DESERIALIZE
 } from 'state/action-types';
 
 import { itemsSchema } from './schema';
 
-export const items = createReducer(
-	{},
-	{ [ SITE_UPDATES_RECEIVE ]: ( state, { siteId, updates } ) => Object.assign( {}, state, { [ siteId ]: updates } ) },
-	itemsSchema
-);
+const receiveUpdatesForSites = ( state, sites ) => {
+	return reduce( sites, ( memo, site ) => {
+		if ( memo === state ) {
+			memo = { ...state };
+		}
```

Comment:
I assume this copy is made here rather than in `reduce`'s last argument so that we don't create another instance of `state` if `sites` is empty? If so, this will not prevent the following case:
- `sites` is not empty
- `sites` does not contain any `site` where `site.updates`
- return value is now a different instance, even though it is equal to previous `state`

I would suggest maybe something like:

```js
const receiveUpdatesForSites = ( state, sites ) => {
  const updatedSites = sites.filter( ( site ) => site.updates );
  return isEmpty( updatedSites )
    ? state
    : sites.reduce( ( newState, site ) => {
        newState[ site.ID ] = site.updates;
      }, { ...state } );
}
```